### ISSUE TITLE
[BUG] - Dropdown Causes Cascading close without mouse event trigger when nested in a Popover

### ISSUE THREAD

**Issue by buchananwill** [nextui-org/nextui#3551]
### NextUI Version

2.4.1

### Describe the bug

✅ Open Popover
✅ Open Dropdown
✅ Select/Deselect items with keyboard
❌ Click an item and everything closes without event listeners firing

This code is a minimum reproduction. I tried adding a breakpoint in Chrome for mouse events, but it doesn't fire anything at all when you try to click a dropdown item. Up to that point the breakpoints are firing/pausing correctly.

UPDATE: Behaviour is identical in Edge.

```
'use client';
import { Popover, PopoverContent, PopoverTrigger } from '@nextui-org/popover';
import { Button } from '@nextui-org/button';
import React, {
  Dispatch,
  SetStateAction,
  useCallback,
  useMemo,
  useState
} from 'react';
import {
  Dropdown,
  DropdownItem,
  DropdownMenu,
  DropdownTrigger,
  Selection
} from '@nextui-org/react';
import { Identifier } from 'dto-stores';
import { ChevronDownIcon } from '@heroicons/react/24/outline';

export default function Page() {
  const [visibleColumns, setVisibleColumns] = useState<Selection>(
    new Set(['type'] as Identifier[])
  );

  const setVisibleColumnsIntercept: DispatchState<Selection> = useCallback(
    (selectionDispatch: SetStateAction<Selection>) => {
      console.log('Using callback: ', selectionDispatch);
      setVisibleColumns(selectionDispatch);
    },
    []
  );

  const headerColumns = useMemo(() => {
    console.log('setting visible columns:', visibleColumns);
    if (visibleColumns === 'all') return columns;

    return columns.filter((column) => visibleColumns.has(column.uid));
  }, [visibleColumns]);

  return (
    <Popover shouldCloseOnBlur={false}>
      <PopoverTrigger>
        <Button>Open</Button>
      </PopoverTrigger>
      <PopoverContent>
        <Dropdown
          shouldCloseOnBlur={false}
          shouldCloseOnInteractOutside={() => false}
        >
          <DropdownTrigger className="hidden sm:flex">
            <Button
              endContent={<ChevronDownIcon className="text-small" />}
              variant="flat"
            >
              Columns
            </Button>
          </DropdownTrigger>
          <DropdownMenu
            disallowEmptySelection
            aria-label="Table Columns"
            closeOnSelect={false}
            shouldFocusWrap={true}
            items={columns}
            selectedKeys={visibleColumns}
            selectionMode="multiple"
            onSelectionChange={setVisibleColumns}
          >
            {(item) => <DropdownItem key={item.uid}>{item.name}</DropdownItem>}
          </DropdownMenu>
        </Dropdown>
      </PopoverContent>
    </Popover>
  );
}

interface Pet {
  type: string;
  name: string;
  id: number;
}

const columns: Column<Pet>[] = [
  { uid: 'type', name: 'Type', sortable: true },
  { uid: 'id', name: 'Id', sortable: true },
  { uid: 'name', name: 'Name', sortable: true }
];

const pets: Pet[] = [
  { type: 'Cat', id: 1, name: 'Amber' },
  { type: 'Cat', id: 2, name: 'Lincoln' },
  { type: 'Guinea Pig', id: 3, name: 'Cotton Bud' },
  { type: 'Rabbit', id: 4, name: 'Dandelion' }
];

type DispatchState<T> = Dispatch<SetStateAction<T>>;

interface Column<T> {
  name: string;
  uid: Extract<keyof T, string | number>;
  sortable?: boolean;
}

```

### Steps to Reproduce the Bug or Issue

As  described above.

### Expected behavior

Dropdowns should equally accept mouse or keyboard events when nested in Popovers.

### Screenshots or Videos

![dropdown nesting](https://github.com/user-attachments/assets/33f55678-e090-41d1-bdcb-e0cbf0765bce)

### Operating System Version

- Windows 10 Home V.22H2

### Browser

Chrome

**Comment by wingkwong** [comment]
Please try to use the latest version (v2.4.6) to check if the issue is still reproducible. I remember we've made some changes for that.

**Comment by buchananwill** [comment]
Aha! Yes the bug is fixed in version 2.4.6. Thanks!
